components/collections/codepointtrie_builder/cpp/list_to_ucptrie.cpp

```diff
@@ -22,9 +25,37 @@ extern "C" {
  * n: The length of values array.
  */
 int construct_ucptrie(const uint32_t defaultValue, const uint32_t errorValue,
-                      const UCPTrieType trieType,
-                      const UCPTrieValueWidth valueWidth,
-                      const uint32_t* values, const uint32_t n) {
+                      const uint32_t trieTypeUint,
+                      const uint32_t valueWidthUint, const uint32_t* values,
+                      const uint32_t n) {
+  UCPTrieType trieType;
```

Comment:
Can't you just do `(UCPTrieType) trieTypeUint`?

Reply:
I have a slight preference for the explicit switch statement as currently written.

Reply:
Moved to using `static_cast`